### ISSUE TITLE
error-is-as: ErrorAs warning when using generic target

### ISSUE THREAD

**Issue by uudashr** [Antonboom/testifylint#213]
Below code works, but it shows warning

> error-is-as: second argument to require.ErrorAs must be a non-nil pointer to either a type that implements error, or to any interface type (testifylint) go-golangci-lint

```go
package requirekit

import (
  "github.com/stretchr/testify/require"
)

func ErrorAs[T error]() require.ErrorAssertionFunc {
  return func(t require.TestingT, err error, msgAndArgs ...interface{}) {
    var target *T
    require.ErrorAs(t, err, &target, msgAndArgs...)
  }
}
```

**Comment by Antonboom** [comment]
@uudashr hi!

this is not a false positive

you should remove extra `*` in `var target *T`:

```go
func ErrorAs[T error]() require.ErrorAssertionFunc {
    return func(t require.TestingT, err error, msgAndArgs ...interface{}) {
        var target T
        require.ErrorAs(t, err, &target, msgAndArgs...)
    }
}
```

More examples – https://github.com/search?q=func+ErrorAs%5BT+error%5D&type=code.
Also, just try to run `go vet` on your code or run the test – you should receive the error from `errors.As` itself.

But the code above panics on assertion failure, because of nil ptr dereference here:

https://github.com/stretchr/testify/blob/7c367bb7bc7dad2d8b17921a604157dc27ebe06f/assert/assertions.go#L2178
(`err.Error()` call)

Related to https://github.com/stretchr/testify/issues/1343
